contracts/staking/ERC20TokenStakingManager.sol

```diff
@@ -50,18 +56,35 @@ contract ERC20TokenStakingManager is Initializable, StakingManager, IERC20TokenS
     }
 
     function initialize(
-        StakingManagerSettings calldata settings,
+        ValidatorManagerSettings calldata settings,
+        uint256 minimumStakeAmount,
+        uint256 maximumStakeAmount,
+        uint64 minimumStakeDuration,
+        IRewardCalculator rewardCalculator,
```

Comment:
It may be cleaner to combine these into a `PoSValidatorManagerSettings` in `IPoSValidatorManager.sol`. That way all PoS validator managers can use the same type to initialize.